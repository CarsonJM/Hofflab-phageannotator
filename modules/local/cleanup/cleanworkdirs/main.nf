process CLEANWORKDIRS {
    tag "${meta.id}|${module_being_cleaned}"
    label "process_single"

    input:
    tuple val(meta), val(directory), val(module_being_cleaned)

    output:
    tuple val(meta), path("*.log")  , emit: cleaned_workdir

    script:
    def files_to_clean = ""
        nextflow_files = [".command.begin", ".command.err", ".command.log", ".command.out", ".command.run", ".command.sh", ".command.trace", ".exitcode", "versions.yml"]
        file(directory, checkIfExists:true).eachFileRecurse { file ->
            files_to_clean = nextflow_files.any { file.name =~ it } ? files_to_clean : files_to_clean + " " + file
        }
    """
    for file in${files_to_clean}; do
        if ! [ -L \$file ]; then
            clean_work_files.sh "\$file" "null" >> ${meta.id}.${module_being_cleaned}.log 2>&1
        fi
    done
    """

    stub:
    def files_to_clean = ""
        nextflow_files = [".command.begin", ".command.err", ".command.log", ".command.out", ".command.run", ".command.sh", ".command.trace", ".exitcode", "versions.yml"]
        file( directory, checkIfExists:true ).eachFileRecurse { file ->
            files_to_clean = nextflow_files.any { file.name =~ it } ? files_to_clean : files_to_clean + " " + file
        }
    """
    touch ${meta.id}.${module_being_cleaned}.log
    """
}
