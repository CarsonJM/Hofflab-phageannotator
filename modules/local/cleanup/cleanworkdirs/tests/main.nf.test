nextflow_process {

    name "Test Process: CLEANWORKDIRS"
    script "../main.nf"
    process "CLEANWORKDIRS"

    setup {
        run("LOGAN_DOWNLOADFILTERTRFINDER") {
            script "../../../logan/downloadfiltertrfinder/main.nf"
            process {
                """
                input[0] = Channel.fromFilePairs("${projectDir}/assets/test-datasets/samplesheets/logan.csv", size: 1 )
                    .map { meta, accession_file ->
                        [ [ id: meta ] , accession_file ]
                    }
                """
            }
        }
    }

    test("logan workDir") {
        when {
            process {
                """
                input[0] = LOGAN_DOWNLOADFILTERTRFINDER.out.fasta
                    .map { meta, fasta ->
                        [ meta, fasta.getParent().toString(), "LOGAN_DOWNLOADFILTERTRFINDER" ]
                    }
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(file(process.out.cleaned_workdir[0][1]).name).match() }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = LOGAN_DOWNLOADFILTERTRFINDER.out.fasta
                    .map { meta, fasta ->
                        [ meta, fasta.getParent().toString(), "LOGAN_DOWNLOADFILTERTRFINDER" ]
                    }
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
