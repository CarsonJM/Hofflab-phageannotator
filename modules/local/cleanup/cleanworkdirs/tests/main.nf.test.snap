{
    "logan workDir": {
        "content": [
            "logan.LOGAN_DOWNLOADFILTERTRFINDER.log"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-20T13:38:41.428436321"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "logan"
                        },
                        "logan.LOGAN_DOWNLOADFILTERTRFINDER.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cleaned_workdir": [
                    [
                        {
                            "id": "logan"
                        },
                        "logan.LOGAN_DOWNLOADFILTERTRFINDER.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-20T13:38:51.892250846"
    }
}